pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -35,32 +36,22 @@ type VMsMutator struct {
 	ClusterConfig *virtconfig.ClusterConfig
 }
 
-// until the minimum supported version is kubernetes 1.15 (see https://github.com/kubernetes/kubernetes/commit/c2fcdc818be1441dd788cae22648c04b1650d3af#diff-e057ec5b2ec27b4ba1e1a3915f715262)
-// the mutating webhook must pass silently on errors instead of returning errors
-func emptyValidResponse() *admissionv1.AdmissionResponse {
-	return &admissionv1.AdmissionResponse{
-		Allowed: true,
-	}
-}
-
 func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
 	if !webhookutils.ValidateRequestResource(ar.Request.Resource, webhooks.VirtualMachineGroupVersionResource.Group, webhooks.VirtualMachineGroupVersionResource.Resource) {
-		log.Log.V(1).Warningf("vm-mutator: received invalid request")
-		return emptyValidResponse()
+		err := fmt.Errorf("expect resource to be '%s'", webhooks.VirtualMachineGroupVersionResource.Resource)
+		return webhookutils.ToAdmissionResponseError(err)
```

Comment:
I don't know much about this webhook, but this change of semantics seems important and testable. If I'm correct, can you add a "negative" test for it?

Reply:
thanks.